docs/rotating-keys.md

```diff
@@ -0,0 +1,211 @@
+        new FrozenClock(new DateTimeImmutable('2023-11-04 21:06:35+00:00'))
+    )
+];
+
+$jwt = ''; // Fetched from, for example, a request header
+
+$token = (new JwtFacade())->parse($jwt, ...$validationConstraints);
+```
+
+### Performing a backwards compatible rotation
+
+Now Imagine that you want to adopt the new `BLAKE2B` symmetric algorithm.
+
+These are the changes to your issuing logic:
+
+```diff
+ <?php
+ declare(strict_types=1);
+ 
+ namespace MyApp;
+ 
+ require 'vendor/autoload.php';
+ 
+ use DateTimeImmutable;
+ use Lcobucci\Clock\FrozenClock;
+ use Lcobucci\JWT\Builder;
+ use Lcobucci\JWT\JwtFacade;
+ use Lcobucci\JWT\Signer;
+ use Lcobucci\JWT\Signer\Key\InMemory;
+ 
+ // `FrozenClock` is used here to fix to a point in time that allows our validation to pass
+ $clock = new FrozenClock(new DateTimeImmutable('2023-11-04 21:06:01+00:00')); 
+ $token = (new JwtFacade(clock: $clock))->issue(
+-    new Signer\Hmac\Sha256(),
++    new Signer\Blake2b(),
+-    InMemory::plainText(
+-        'a-very-long-and-secure-key-that-should-actually-be-something-else'
++    InMemory::base64Encoded(
++        'GOu4rLyVCBxmxP+sbniU68ojAja5PkRdvv7vNvBCqDQ='
+     ),
+     static fn (Builder $builder): Builder => $builder
+         ->issuedBy('https://api.my-awesome-app.io')
+         ->permittedFor('https://client-app.io')
+ );
+```
+
+!!! Sample
+    Here's a token issued with the code above, if you want to test the script locally:
+
+    <details>
+        <summary>Sample token</summary>
+        
+        // line breaks added for readability
+        eyJ0eXAiOiJKV1QiLCJhbGciOiJCTEFLRTJCIn0
+        .eyJpYXQiOjE2OTkxMzE5NjEsIm5iZiI6MTY5OTEzMTk2MSwiZXhwIjoxNjk5MTMyMjYxLCJpc3Mi
+        OiJodHRwczovL2FwaS5teS1hd2Vzb21lLWFwcC5pbyIsImF1ZCI6Imh0dHBzOi8vY2xpZW50LWFwc
+        C5pbyJ9.bD67s8IXpAJiBTIZn1et_M5WSS7kfmuNiacNRz5lArQ
+    </details>
+
+So far, nothing different that a normal rotation.
+
+Now check the changes on the parsing and validation logic:
+
+```diff
+ <?php
+ declare(strict_types=1);
+ 
+ namespace MyApp;
+ 
+ require 'vendor/autoload.php';
+ 
+ use DateTimeImmutable;
+ use Lcobucci\Clock\FrozenClock;
+ use Lcobucci\JWT\JwtFacade;
+ use Lcobucci\JWT\Signer;
+ use Lcobucci\JWT\Signer\Key\InMemory;
+ use Lcobucci\JWT\Validation\Constraint;
+ 
+ $validationConstraints = [
+-    new Constraint\SignedWith(
+-        new Signer\Hmac\Sha256(),
+-        InMemory::plainText(
+-            'a-very-long-and-secure-key-that-should-actually-be-something-else'
+-        ),
+-    ),
++    new Constraint\SignedWithOneInSet(
++       new Constraint\SignedWith(
++           new Signer\Blake2b(),
++           InMemory::base64Encoded(
++               'GOu4rLyVCBxmxP+sbniU68ojAja5PkRdvv7vNvBCqDQ='
++           ),
++       ),
++       new Constraint\SignedWith(
++            new Signer\Hmac\Sha256(),
++           InMemory::plainText(
++                'a-very-long-and-secure-key-that-should-actually-be-something-else'
++           ),
++       ),
++    ),
```

Comment:
Correct me if I'm wrong, the only foreseeable valid use for `SignedWithOneInSet` is key rotation.

If so, I would like to protect the developer to forget to delete the old key once all the tokens signed with that key have expired.

For example, I'd like to see a bind between each key and a time constraint:

1. `SignedWith(Sha256()) ` -> valid until `2023-11-30 00:00:00`
1. `SignedWith(Blake2b())` -> valid until `2025-11-30 00:00:00`

Not sure about the UX and implementation details, but please let's avoid adding code in production that gets forgotten leaving a possible weakness opened forever

Reply:
That's a good point, thanks! 
I need to think about the interface here... A possibility would be defining an expiration date to the key itself.

Reply:
Creating that bond complicates the design a little bit, especially because we don't want to have it as an optional thing.

I'll put an intermediate object to represent that set and see where it takes us.

Reply:
@Slamdunk I've pushed [26960f1](https://github.com/lcobucci/jwt/pull/1033/commits/26960f17cc8040dea553b3fb9fa6c02d86ea2342) with the experimental design.

What are your thoughts?

Reply:
Awesome :muscle: 

Only doc is missing the update, then we can go :+1:

Reply:
Done!